pkg/sdn/apis/network/validation/validation_test.go

```diff
@@ -156,55 +145,50 @@ func TestSetDefaultClusterNetwork(t *testing.T) {
-				HostSubnetLength: 8,
-				ServiceNetwork:   "172.30.0.0/16",
-				PluginName:       "redhat/openshift-ovs-multitenant",
+				ObjectMeta:      metav1.ObjectMeta{Name: sdnapi.ClusterNetworkDefault},
+				ClusterNetworks: []sdnapi.ClusterNetworkEntry{{CIDR: "10.30.0.0/16", HostSubnetLength: 8}},
+				ServiceNetwork:  "172.30.0.0/16",
+				PluginName:      "redhat/openshift-ovs-multitenant",
 			},
 			expectedErrors: 1,
 		},
 		{
 			name: "Additional Network",
 			cn: &sdnapi.ClusterNetwork{
-				ObjectMeta:       metav1.ObjectMeta{Name: sdnapi.ClusterNetworkDefault},
-				ClusterNetworks:  []sdnapi.ClusterNetworkEntry{{CIDR: "10.20.0.0/16"}, {CIDR: "10.30.0.0/16"}},
-				HostSubnetLength: 8,
-				ServiceNetwork:   "172.30.0.0/16",
-				PluginName:       "redhat/openshift-ovs-multitenant",
+				ObjectMeta:      metav1.ObjectMeta{Name: sdnapi.ClusterNetworkDefault},
+				ClusterNetworks: []sdnapi.ClusterNetworkEntry{{CIDR: "10.20.0.0/16", HostSubnetLength: 8}, {CIDR: "10.30.0.0/16", HostSubnetLength: 8}},
+				ServiceNetwork:  "172.30.0.0/16",
+				PluginName:      "redhat/openshift-ovs-multitenant",
 			},
 			expectedErrors: 1,
 		},
 		{
 			name: "Wrong HostSubnetLength",
 			cn: &sdnapi.ClusterNetwork{
-				ObjectMeta:       metav1.ObjectMeta{Name: sdnapi.ClusterNetworkDefault},
-				ClusterNetworks:  []sdnapi.ClusterNetworkEntry{{CIDR: "10.20.0.0/16"}},
-				HostSubnetLength: 9,
-				ServiceNetwork:   "172.30.0.0/16",
-				PluginName:       "redhat/openshift-ovs-multitenant",
+				ObjectMeta:      metav1.ObjectMeta{Name: sdnapi.ClusterNetworkDefault},
+				ClusterNetworks: []sdnapi.ClusterNetworkEntry{{CIDR: "10.20.0.0/16", HostSubnetLength: 9}},
+				ServiceNetwork:  "172.30.0.0/16",
+				PluginName:      "redhat/openshift-ovs-multitenant",
 			},
 			expectedErrors: 1,
 		},
 		{
 			name: "Wrong ServiceNetwork",
 			cn: &sdnapi.ClusterNetwork{
-				ObjectMeta:       metav1.ObjectMeta{Name: sdnapi.ClusterNetworkDefault},
-				ClusterNetworks:  []sdnapi.ClusterNetworkEntry{{CIDR: "10.20.0.0/16"}},
-				HostSubnetLength: 8,
-				ServiceNetwork:   "172.20.0.0/16",
-				PluginName:       "redhat/openshift-ovs-multitenant",
+				ObjectMeta:      metav1.ObjectMeta{Name: sdnapi.ClusterNetworkDefault},
+				ClusterNetworks: []sdnapi.ClusterNetworkEntry{{CIDR: "10.20.0.0/16", HostSubnetLength: 8}},
+				ServiceNetwork:  "172.20.0.0/16",
+				PluginName:      "redhat/openshift-ovs-multitenant",
 			},
 			expectedErrors: 1,
 		},
 		{
 			name: "Wrong PluginName",
 			cn: &sdnapi.ClusterNetwork{
-				ObjectMeta:       metav1.ObjectMeta{Name: sdnapi.ClusterNetworkDefault},
-				ClusterNetworks:  []sdnapi.ClusterNetworkEntry{{CIDR: "10.20.0.0/16"}},
-				HostSubnetLength: 8,
-				ServiceNetwork:   "172.30.0.0/16",
-				PluginName:       "redhat/openshift-ovs-subnet",
+				ObjectMeta:      metav1.ObjectMeta{Name: sdnapi.ClusterNetworkDefault},
+				ClusterNetworks: []sdnapi.ClusterNetworkEntry{{CIDR: "10.20.0.0/16", HostSubnetLength: 8}},
```

Comment:
This section is testing the ValidateClusterNetwork which validates the default clusternetwork and ensures that you cannot modify the values of the default clusternetwork via the api. I did add a test which adds attempts to add a network which fails and there was one that tries to change the hostsubnet field witch also fails. Do you want me to add another test to ValidateClusterNetwork which tests if the default ClusterNetwork starts with multiple cidr/hostsubnetlengh? There are tests for parsing mutliple cidr/hostsubnet in pkg/sdn/plugin/common_test.go.